File: ReviewService/src/main/java/com/clarivate/reviewservice/Service/EditorService.java
package com.clarivate.reviewservice.Service;

import com.clarivate.reviewservice.dto.Request.AssignReviewerRequest;
import com.clarivate.reviewservice.dto.Request.EditorDecisionRequest;
import com.clarivate.reviewservice.dto.Request.ReviewCommentRequest;
import com.clarivate.reviewservice.dto.Response.ReviewCommentResponse;
import com.clarivate.reviewservice.dto.Response.ReviewProcessResponse;

import java.util.List;

public interface EditorService {
    ReviewProcessResponse assignReviewer(AssignReviewerRequest request);
    ReviewProcessResponse makeFinalDecision(Long reviewId, EditorDecisionRequest request);
    List<ReviewProcessResponse> getPendingReviews();
    List<ReviewProcessResponse> getAssignedReviews(Long editorId);
    ReviewProcessResponse getReview(Long reviewId);
}

File: research-frontend-main/src/app/core/services/editor.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '@env/environment';
import { AssignReviewerRequest, AvailableReviewerResponse, EditorDecisionRequest, ReviewProcessResponse } from '@core/models';

@Injectable({ providedIn: 'root' })
export class EditorService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrls.review}/editor`;

  getPendingReviews(): Observable<ReviewProcessResponse[]> {
    return this.http.get<ReviewProcessResponse[]>(`${this.baseUrl}/reviews/pending`);
  }

  getAvailableReviewers(): Observable<AvailableReviewerResponse[]> {
    return this.http.get<AvailableReviewerResponse[]>(`${this.baseUrl}/reviewers`);
  }

  assignReviewer(request: AssignReviewerRequest): Observable<ReviewProcessResponse> {
    return this.http.post<ReviewProcessResponse>(`${this.baseUrl}/assign-reviewer`, request);
  }

  makeFinalDecision(reviewId: number, request: EditorDecisionRequest): Observable<ReviewProcessResponse> {
    return this.http.put<ReviewProcessResponse>(`${this.baseUrl}/reviews/${reviewId}/decision`, request);
  }

  getAssignedReviews(editorId: number): Observable<ReviewProcessResponse[]> {
    return this.http.get<ReviewProcessResponse[]>(`${this.baseUrl}/${editorId}/reviews`);
  }

  getReview(reviewId: number): Observable<ReviewProcessResponse> {
    return this.http.get<ReviewProcessResponse>(`${this.baseUrl}/reviews/${reviewId}`);
  }
}
